services/account.service.js

```diff
@@ -20,6 +21,30 @@ function findById(id) {
     );
 }
 
+/**
+ * @function findByHackerId
+ * @param {ObjectId} id the Hacker's ID
+ * @returns {Promise<Account>} The account of the hacker, minus the password. Returns null if the hacker does not exist, or if the hacker is not associated with an account.
+ * Get the account by using the hacker's ID.
+ */
+async function findByHackerId(id) {
+    const TAG = `[Account Service # findByHackerId]:`;
+    const query = {
+        _id: id
+    };
+    const hacker = await Hacker.findById(
+        query,
+        logger.queryCallbackFactory(TAG, "account", query)
+    ).populate({
+        path: "accountId",
+        select: " -password"
```

Comment:
what does this do?

Reply:
Selects all fields except for the password. We never want to populate the password unless we're doing a login / auth.